### PR TITLE
refactor: refactor dialog to use scss

### PR DESCRIPTION
**Pull request by noahwaldner**
- refactor styles to scss
- add mixin for sm media query

### CODE REVIEW

File: packages/components/src/components/Dialog/styles/dialog.module.scss
Comment (SamuelAlev) [inline packages/components/src/components/Dialog/styles/dialog.module.scss:1]: I just opened the first story (https://deploy-preview-1906--fondue-components.netlify.app/?path=/story/current_components-dialog--with-full-layout) and there are inconsistencies with main (https://fondue-components.frontify.com/?path=/story/current_components-dialog--with-full-layout)

Please check over one by one.

File: packages/components/package.json [hunk 1]
```diff
@@ -71,6 +71,7 @@
         "@storybook/addon-links": "^8.1.5",
         "@storybook/blocks": "^8.1.5",
         "@storybook/manager-api": "^8.1.5",
+        "@storybook/preset-scss": "^1.0.3",
```
Comment (SamuelAlev) [inline packages/components/package.json:74]: vite support scss by default and the plugin was necessary when storybook was using webpack instead of vite

Comment (noahwaldner) [reply packages/components/package.json:74]: removed, it was a chache issue

**Comment by noahwaldner** [comment]
Styles adjusted to match tw version

File: packages/components/package.json (hunk 1 above)
Comment (SamuelAlev) [inline packages/components/package.json:74]: Still there

File: packages/components/package.json [hunk 2]
```diff
@@ -93,7 +94,6 @@
         "postcss-import": "^16.1.0",
         "prettier": "^3.3.0",
         "react": "^18.3.1",
-        "react-dom": "^18.3.1",
```
Comment (SamuelAlev) [inline packages/components/package.json:96]: Please revert back

**Pull request merged**